src/UriHelper.php

```diff
@@ -179,26 +193,34 @@ public static function getSrcSetUrlString($url, $size, $custom = null)
      * This method will then generate the right rokka URLs to get what you want, see
      * `\Rokka\Client\Tests\UriHelperTest::provideGetSrcSetUrl` for some examples and the expected returns.
      *
-     * @param UriInterface $url    The original rokka render URL to be adjusted
-     * @param string       $size   The size of the image, eg '300w' or '2x'
-     * @param null|string  $custom Any rokka options you'd like to add, or are a dpi identifier like '2x'
+     * @param UriInterface $url           The original rokka render URL to be adjusted
+     * @param string       $size          The size of the image, eg '300w' or '2x'
+     * @param null|string  $custom        Any rokka options you'd like to add, or are a dpi identifier like '2x'
+     * @param bool         $setWidthInUrl If false, don't set the width as stack operation option, we provide it in $custom, usually as parameter
+     *
+     * @throws \RuntimeException
      *
      * @return UriInterface
      */
-    public static function getSrcSetUrl(UriInterface $url, $size, $custom = null)
+    public static function getSrcSetUrl(UriInterface $url, $size, $custom = null, $setWidthInUrl = true)
     {
         $identifier = substr($size, -1, 1);
         $size = substr($size, 0, -1);
         if ('x' === $identifier) {
             $uri = self::addOptionsToUri($url, 'options-dpr-'.$size);
         } elseif ('w' === $identifier) {
-            $uri = self::addOptionsToUri($url, 'resize-width-'.$size);
+            if ($setWidthInUrl) {
+                $uri = self::addOptionsToUri($url, 'resize-width-'.$size);
+            } else {
+                $uri = $url;
+            }
         } else {
             return $url;
         }
         if (null !== $custom) {
```

Comment:
This function is long and a bit difficult to read, maybe the custom part could be a private method to make it a bit more clear?

Reply:
moved